src/Algebra/Graph/NonEmpty.hs

```diff
@@ -681,7 +699,14 @@ transpose = foldg1 vertex overlay (flip connect)
 -- induce1 p '>=>' induce1 q == induce1 (\\x -> p x && q x)
 -- @
 induce1 :: (a -> Bool) -> NonEmptyGraph a -> Maybe (NonEmptyGraph a)
-induce1 p = toNonEmptyGraph . G.induce p . T.toGraph
+induce1 p = foldg1
+  (\x -> if p x then Just (Vertex x) else Nothing)
+  (k Overlay)
```

Comment:
Isn't this just `liftA2 Overlay` and `liftA2 Connect`?

Reply:
Totally, copying/pasting is a bad thing :)

Reply:
Well, no it is not, because `liftA2 (+) (Nothing) (Just 1) == Nothing`, we don't want this behavior

Reply:
Oops, you are right :)